src/lock.rs

```diff
@@ -158,21 +142,6 @@ impl<'de> Deserialize<'de> for Digest {
     }
 }
 
-impl Ord for Digest {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        match self.algorithm.cmp(&other.algorithm) {
-            std::cmp::Ordering::Equal => self.bytes.cmp(&other.bytes),
-            other => other,
-        }
-    }
-}
-
-impl PartialOrd for Digest {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(self.cmp(other))
-    }
-}
-
```

Comment:
Similar here, this was done so ordering was done thoughtfully. It's particularly important here because the algorithm type has higher precedence than the bytes content when comparing digests, and you don't get this automatically from the default implementation.

Reply:
You would get the same implementation if you change the order of the fields. Meaning that if you make it like this:

```rust
struct Digest {
    algorithm: DigestAlgorithm,
    bytes: Vec<u8>,
}
```

Unless there is a reason not to order the fields like this, I would strongly encourage using `derive(Ord, PartialOrd)` because it is a lot more maintainable. Too easy to add fields and forget to update the hand-rolled implementations. Let the compiler do the work!

Reply:
Yes, this is also a valid solution. If you add a comment about the importance of the field ordering it should be fine to use derivations.

Reply:
Sweet!